.github/workflows/trivy.yml

```diff
@@ -0,0 +1,57 @@
+name: Trivy Scanner
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 8 * * *" # runs at 8 AM UTC, 4 AM EST.
+
+jobs:
+  trivy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Build all projects without running tests
+        run: ./gradlew --build-cache build -x test -x spotlessCheck
+
+      - name: Construct docker image name and tag
+        id: image-name
+        run: |
+          GITHUB_REPO=$(basename ${{ github.repository }})
+          GIT_SHORT_HASH=$(git rev-parse --short HEAD)
+          echo ::set-output name=name::${GITHUB_REPO}:${GIT_SHORT_HASH}
+
+      - name: Build image locally with jib
```

Comment:
Could this instead run the scan on the latest built image in artifactory or gcr? What is the benefit of building it here instead of getting it from there? In a way, I think scanning the published image is more important since it doesn't matter if it builds and scans if the published copy is invalid.

Maybe something like
```
        run: docker pull gcr.io/broad-dsp-gcr-public/terra-data-catalog
```

Reply:
I went back to our production readiness documentation and for compliance and security it says: "Trivy is set up as a GitHub Action and runs against **each PR**". I need to rework this action.

Reply:
Additionally, I'm not sure how useful it is to scan existing images: this will not help us be proactive about upcoming issues, and will only tell us existing issues which might already be fixed.

Reply:
> I went back to our production readiness documentation and for compliance and security it says: "Trivy is set up as a GitHub Action and runs against each PR". I need to rework this action.

OK, if we _are_ scanning on PRs, and an image isn't built yet, then we need to build one here.

> Additionally, I'm not sure how useful it is to scan existing images: this will not help us be proactive about upcoming issues, and will only tell us existing issues which might already be fixed.

I actually think the prebuilt case is more interesting, as it detects newly discovered issues in images that we have already published, and are running in production. But my basic problem is scanning an image that we build on the fly. Also, we have no way to assert that what we built is the same as the image that's used when running our service.